ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/BadBlockManager.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.chain;
+
+import org.hyperledger.besu.ethereum.core.Block;
+import org.hyperledger.besu.ethereum.core.Hash;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class BadBlockManager {
+
+  private final Map<Hash, Block> badBlocks = new ConcurrentHashMap<>();
```

Comment:
We want some sort of a limit on the bad blocks we keep around.  Maybe via `com.google.common.cache.CacheBuilder`?

Reply:
Geth said they keep on the order of 100 in memory.

Reply:
Indeed I wanted to use the cacheBuilder but as I understood that we did not want a limit for this cache I did not set it up. But anyway the choice to put a limit is much better